lib/irb/ruby-lex.rb

```diff
@@ -65,68 +65,43 @@ def configure_io(io)
           else
             false
           end
-        else
+        elsif single_line_command?(code)
           # Accept any single-line input for symbol aliases or commands that transform args
-          next true if single_line_command?(code)
-
-          ltype, indent, continue, code_block_open = check_code_state(code)
-          if ltype or indent > 0 or continue or code_block_open
-            false
-          else
-            true
-          end
+          true
+        else
+          check_target_code, tokens, opens = check_code_state(code)
+          continue = process_continue(tokens)
+          opens.empty? && !continue && !check_code_block(check_target_code, tokens)
         end
       end
     end
     if @io.respond_to?(:dynamic_prompt)
       @io.dynamic_prompt do |lines|
         lines << '' if lines.empty?
-        result = []
         tokens = self.class.ripper_lex_without_warning(lines.map{ |l| l + "\n" }.join, context: @context)
-        code = String.new
-        partial_tokens = []
-        unprocessed_tokens = []
-        line_num_offset = 0
-        tokens.each do |t|
-          partial_tokens << t
-          unprocessed_tokens << t
-          if t.tok.include?("\n")
-            t_str = t.tok
-            t_str.each_line("\n") do |s|
-              code << s
-              next unless s.include?("\n")
-              ltype, indent, continue, code_block_open = check_state(code, partial_tokens)
-              result << @prompt.call(ltype, indent, continue || code_block_open, @line_no + line_num_offset)
-              line_num_offset += 1
-            end
-            unprocessed_tokens = []
-          else
-            code << t.tok
+        line_results = IRB::NestingParser.parse_by_line(tokens)
+        tokens_until_line = []
+        line_results.map.with_index do |(line_tokens, _prev_opens, next_opens), line_num_offset|
```

Comment:
Nitpick: can we add `_min_depth` for clarity? I was confused a bit by the number of params being different 😅